services/py-api/tests/integration_tests/test_participant_routes.py

```diff
@@ -0,0 +1,138 @@
+
+    # Performing cleanup, only the first response is successful
+    await participant_repository.delete(resp1.json()["participant"]["id"])
+
+    assert resp2.status_code == 409
+    assert resp2.json()["error"] == "Participant with this email already exists"
+
+
+@pytest.mark.asyncio(loop_scope="session")
+async def test_create_random_participant_is_admin_true(async_client: AsyncClient) -> None:
+
+    resp = await async_client.post(PARTICIPANT_ENDPOINT_URL, json=IS_ADMIN_TRUE_BODY)
+
+    assert resp.status_code == 422
+    assert resp.json()["detail"][0]["msg"] == "Value error, Field `team_name` is required when `is_admin=True`"
+
+
+@pytest.mark.asyncio(loop_scope="session")
+async def test_create_random_participant_missing_required_fields(async_client: AsyncClient) -> None:
+
+    resp = await async_client.post(PARTICIPANT_ENDPOINT_URL, json=MISSING_REQ_FIELDS_BODY)
+
+    assert resp.status_code == 422
+    assert resp.json()["detail"][0]["type"] == "missing"
+    assert resp.json()["detail"][0]["msg"] == "Field required"
+
+
+@pytest.mark.asyncio(loop_scope="session")
+async def test_create_admin_participant(
+    async_client: AsyncClient, participant_repository: ParticipantsRepository, team_repository: TeamsRepository
+) -> None:
+    resp = await async_client.post(PARTICIPANT_ENDPOINT_URL, json=ADMIN_PARTICIPANT_BODY)
+
+    """
+    >PERFORMING CLEANUP
+    The cleanup is done here so that the failing assertions do not interrupt the cleanup process.
+    We have saved the state of the request in the resp variable and we can still perform checks on it.
+    """
+    await participant_repository.delete(resp.json()["participant"]["id"])
+    await team_repository.delete(resp.json()["team"]["id"])
+
+    assert resp.status_code == 201
+    assert resp.json()["participant"]["name"] == "testtest"
+    assert resp.json()["participant"]["email"] == "testtest@test.com"
+    assert resp.json()["participant"]["is_admin"] == True
+    assert resp.json()["participant"]["email_verified"] == False
+    assert resp.json()["participant"]["team_id"] == resp.json()["team"]["id"]
+    assert resp.json()["team"]["name"] == "testteam"
+    assert resp.json()["team"]["is_verified"] == False
+
+
+@pytest.mark.asyncio(loop_scope="session")
+async def test_create_admin_participant_email_and_team_already_exists(
+    async_client: AsyncClient, participant_repository: ParticipantsRepository, team_repository: TeamsRepository
+) -> None:
+
+    resp1 = await async_client.post(PARTICIPANT_ENDPOINT_URL, json=ADMIN_PARTICIPANT_BODY)
+    resp2 = await async_client.post(PARTICIPANT_ENDPOINT_URL, json=ADMIN_PARTICIPANT_BODY)
+
+    # Performing cleanup, only the first response is successful
+    await participant_repository.delete(resp1.json()["participant"]["id"])
+    await team_repository.delete(resp1.json()["team"]["id"])
+
+    assert resp2.status_code == 409
+    assert resp2.json()["error"] == "Team with this name already exists"
```

Comment:
Nice one, here we are testing if the order of operations is correct (First create team, then Participant). Maybe add a comment about it.

Reply:
Yep right on point, that is why it prints team with this name already exists

Reply:
Fixed in c8eeae7c76c289a8d0700674027b82bc6ec5c972